cdap-common/src/main/java/co/cask/cdap/common/guice/ConfigModule.java

```diff
@@ -57,11 +57,16 @@ public ConfigModule(CConfiguration cConf, Configuration hConf, SConfiguration sC
     this.sConf = sConf;
     CConfigurationUtil.copyTxProperties(cConf, hConf);
 
-    // Set system properties for all HTTP requests
-    System.setProperty(DefaultHttpRequestConfig.CONNECTION_TIMEOUT_PROPERTY_NAME,
-                       cConf.get(Constants.HTTP_CLIENT_CONNECTION_TIMEOUT_MS));
-    System.setProperty(DefaultHttpRequestConfig.READ_TIMEOUT_PROPERTY_NAME,
-                       cConf.get(Constants.HTTP_CLIENT_READ_TIMEOUT_MS));
+    // Set system properties for all HTTP requests if they were found
+    String connectionTimeout = cConf.get(Constants.HTTP_CLIENT_CONNECTION_TIMEOUT_MS);
+    String readTimeout = cConf.get(Constants.HTTP_CLIENT_READ_TIMEOUT_MS);
+
+    if (connectionTimeout != null) {
+      System.setProperty(DefaultHttpRequestConfig.CONNECTION_TIMEOUT_PROPERTY_NAME, connectionTimeout);
```

Comment:
let's add a LOG statement at least, to warn users that their system is misconfigured, somehow